src/app/register/register.component.ts

```diff
@@ -0,0 +1,97 @@
+import { Component, OnInit } from '@angular/core';
+import { RegisterService } from '../services/register/register.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-register',
+  templateUrl: './register.component.html',
+  styleUrls: ['./register.component.scss']
+})
+export class RegisterComponent implements OnInit {
+
+  username: string;
+  password: string;
+  confirmPsw: string;
+  usernameError: string;
+  passwordError: string;
+  confirmPswError: string;
+  messageSuccess: string;
+  messageFail: string;
+
+  constructor(private registerService: RegisterService, private router: Router) { }
+
+  ngOnInit() {
+  }
+
+  CheckUserName(username: string): boolean {
```

Comment:
Maybe ask if the below 4 methods would be better placed in a service rather than in the component itself for re-usability?

Reply:
this methods are left there but only deleted the userExist function. it is implemented into the register function